code/controller/controller.ino

```diff
@@ -148,31 +141,27 @@ void reset() {
 }
 
 #ifdef ENABLE_SERIAL
-void printAnalog(const long& packedVal) {
+void printAnalog(int16_t xval, int16_t yval) {
 
-  int tx = packedVal >> (sizeof(int) * 8);
-  int ty = packedVal & 0xFFFFL;
-  sprintf(msg, "Raw: %lx, x-axis: %d, y-axis: %d", packedVal, tx, ty);
+  snprintf(msg, MSG_LEN-1, "x-axis: %d, y-axis: %d", xval, yval);
   Serial.println(msg);
 }
 #else
-inline void printAnalog(const long& packedVal) {}
+inline void printAnalog(const int16_t xval, const int16_t yval) {}  // shouldn't need this at all, because all printAnalog() should be wrapped; just in case.
 #endif
 
-unsigned long readAnalog(int xPin, int yPin, const int XMIN, const int XMAX, const int YMIN, const int YMAX, bool returnRaw = false) {
-  int readX = analogRead(xPin); // store in temporary variables to use in constrain()
-  int readY = analogRead(yPin); 
+int16_t readAnalog(int pin, const int MIN, const int MAX, bool returnRaw=false) {
+  int16_t ret=0;
+  uint16_t read=0;
   
-  if (!returnRaw) {
-    // clamp values to observed joystick values
-    readX = constrain(readX, XMIN, XMAX);
-    readY = constrain(readY, YMIN, YMAX);
+  read=analogRead(pin);
```

Comment:
Can we do:

if (returnRaw) {
  return read;
}

read = constrain(read, min, max);
return map(read, min, max, -32767, 32767);

Reply:
absolutely.  again, I was keeping the program flow largely unchanged.
i originally had it as 
if(returnRaw) return map(analogRead(pin),min,max));
return map(constrain(analogRead(pin),min,max),min,max,-32767,32767));
but felt that was tooooo much butchering.

Reply:
Don't worry about butchering or keeping the original programm flow. Changes that make the code simpler (or more complex but with some other benefit that makes it worth it) matter more.